int8 pattern
string colour
int8 duration
